### ISSUE TITLE
rbenv install 2.7.1 fails on CentOS 8, not sure what I'm doing wrong.

### ISSUE THREAD

**Issue by tromlet** [rbenv/ruby-build#1495]
Seems like rbenv is correctly installed, but it can't seem to get a version of Ruby installed. Attempting to install Ruby 2.7.1 (and, for that matter, 2.7.2) gives me this log output:

```log
installing rdoc:                    /root/.rbenv/versions/2.7.1/share/ri/2.7.0/system
installing html-docs:               /root/.rbenv/versions/2.7.1/share/doc/ruby
installing capi-docs:               /root/.rbenv/versions/2.7.1/share/doc/ruby
installing command scripts:         /root/.rbenv/versions/2.7.1/bin
installing library scripts:         /root/.rbenv/versions/2.7.1/lib/ruby/2.7.0
installing common headers:          /root/.rbenv/versions/2.7.1/include/ruby-2.7.0
installing manpages:                /root/.rbenv/versions/2.7.1/share/man (man1, man5)
installing default gems from lib:   /root/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0 (build_info, cache, doc, extensions, gems, specifications)
                                    benchmark 0.1.0
/tmp/ruby-build.20201201215342.77081.R4J7qq/ruby-2.7.1/lib/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- openssl (LoadError)
	from /tmp/ruby-build.20201201215342.77081.R4J7qq/ruby-2.7.1/lib/rubygems/core_ext/kernel_require.rb:92:in `require'
	from /tmp/ruby-build.20201201215342.77081.R4J7qq/ruby-2.7.1/lib/rubygems/specification.rb:2426:in `to_ruby'
	from ./tool/rbinstall.rb:846:in `block (2 levels) in install_default_gem'
	from ./tool/rbinstall.rb:279:in `open_for_install'
	from ./tool/rbinstall.rb:845:in `block in install_default_gem'
	from ./tool/rbinstall.rb:835:in `each'
	from ./tool/rbinstall.rb:835:in `install_default_gem'
	from ./tool/rbinstall.rb:799:in `block in <main>'
	from ./tool/rbinstall.rb:950:in `block in <main>'
	from ./tool/rbinstall.rb:947:in `each'
	from ./tool/rbinstall.rb:947:in `<main>'
make: *** [uncommon.mk:373: do-install-all] Error 1
```

Output of `openssl version`: `OpenSSL 1.1.1c FIPS  28 May 2019`
Output of `cat /etc/os-release`:
```
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"
```

Output of `cat ~/.bash_profile`:
```bash
# .bash_profile

# Get the aliases and functions
if [ -f ~/.bashrc ]; then
        . ~/.bashrc
fi

# User specific environment and startup programs

PATH=$PATH:$HOME/bin

export PATH

export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"
```

Output of `cat ~/.bashrc`:
```bash
# .bashrc

# User specific aliases and functions

alias rm='rm -i'
alias cp='cp -i'
alias mv='mv -i'

# Source global definitions
if [ -f /etc/bashrc ]; then
	. /etc/bashrc
fi

# export PATH="$HOME/.rbenv/bin:$PATH"
# eval "$(rbenv init -)"
# export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"
```

**Comment by hsbt** [comment]
@tromlet It's not rbenv issue, maybe ruby-build or your environment. Do not create the duplicate entry rbenv and ruby-build repos.

**Comment by tromlet** [comment]
@hsbt for sure... I actually closed it and moved it to rbenv/rbenv because I figured that was where it should go... but who knows?

I have edited my `~/.bash_profile` - and I verified that I have only one instance of `~/.rbenv/plugins/ruby-build`. Not sure what's going on. :/

**Comment by tnr2323** [comment]
Thank you I’ll figure it out soon just give me a chance

Sent from my iPhone

On Dec 2, 2020, at 7:08 PM, Tom Spettigue <notifications@github.com> wrote:

﻿
@hsbt for sure... I actually closed it and moved it to rbenv/rbenv because I figured that was where it should go... but who knows?

I have edited my ~/.bash_profile - and I verified that I have only one instance of ~/.rbenv/plugins/ruby-build. Not sure what's going on. :/

**Comment by tromlet** [comment]
So I fixed it. Turns out the package `openssl-devel` was not installed on my system (I start all my servers from a CentOS "Minimal" installation), which is why it was getting hung up on that openssl part of... whatever it was doing there. I installed it with a simple `dnf install openssl-devel`.

I had also fully uninstalled rbenv per the instructions at https://github.com/rbenv/rbenv#uninstalling-rbenv, rebooted, and reinstalled it by following the instructions at https://github.com/rbenv/rbenv#basic-github-checkout (step 4 was completed by running the `source ~/.bash_profile` command). After the rbenv-doctor script *DIDN'T* find `rbenv install` support, I had to pop over to https://github.com/rbenv/ruby-build#installation to get `ruby-build` all set up and installed.

After that, my `rbenv install` command in my rails git repo worked just great. :)
